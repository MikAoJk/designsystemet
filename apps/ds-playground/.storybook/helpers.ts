export const category = {
  baseProps: 'BaseProps',
  props: 'Props',
  htmlAttribute: 'HTML-attribute',
  aria: 'Aria-attribute',
  event: 'Event',
};
export const htmlEventDescription = {
  table: { category: category.event },
  control: false,
};

/* export const getArgsWithCategory = <T>({
  storyProps,
  categories,
}: {
  storyProps: any;
  categories: string[];
}): { [key: string]: undefined } => {
  const argsObj: { [key: string]: undefined } = {};
  categories.forEach((e) => {
    const foundcategory = Object.entries(storyProps.argTypes).find(
      ([, value]) => {
        return value.table.category === e;
      }
    );
    if (foundcategory) argsObj[foundcategory[0]] = undefined;
  });
  return argsObj;
};
 */

export const getArgsWithCategory = (): { [key: string]: undefined } => {
  return { ettellerannet: undefined };
};
