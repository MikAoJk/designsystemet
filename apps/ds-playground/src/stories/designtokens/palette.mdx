import { Meta, Story, Canvas } from '@storybook/blocks';
import { PaletteTokenExamplesGenerator } from './PaletteTokenExamplesGenerator';
import * as PaletteStories from './palette.stories';
import { VersionBadge } from '../utils/VersionBadge';

# Palette

<VersionBadge packageName={'core-designtokens'} />

<Meta of={PaletteStories} />

<PaletteTokenExamplesGenerator />

<Canvas withSource="none">
  <Story of={PaletteStories.Palette} />
</Canvas>
