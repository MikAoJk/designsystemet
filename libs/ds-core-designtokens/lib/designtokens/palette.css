:root,
:host {
  --palette-burgundy-100: '#6f2c3f';
  --palette-burgundy-70: '#b04c5c';
  --palette-burgundy-50: '#f06674';
  --palette-burgundy-30: '#f3a7b0';
  --palette-burgundy-10: '#f7e2e8';
  --palette-burgundy-5: '#faf0f3';
  --palette-forest-100: '#2b6940';
  --palette-forest-70: '#5d9b73';
  --palette-forest-50: '#91d6ac';
  --palette-forest-30: '#b9e1c8';
  --palette-forest-10: '#e3f5ea';
  --palette-forest-5: '#f4faf5';
  --palette-ochre-100: '#6b492c';
  --palette-ochre-70: '#a9805b';
  --palette-ochre-50: '#e7b78a';
  --palette-ochre-30: '#f0d2b6';
  --palette-ochre-10: '#f9ede2';
  --palette-ochre-5: '#f9f4f0';
  --palette-denim-100: '#2c436b';
  --palette-denim-70: '#5a77a8';
  --palette-denim-50: '#89abe5';
  --palette-denim-30: '#b5cbee';
  --palette-denim-10: '#e2eaf7';
  --palette-denim-5: '#eff3f9';
  --palette-azure-100: '#093e61';
  --palette-azure-70: '#1362ae';
  --palette-azure-30: '#9ccff2';
  --palette-azure-10: '#cde1f9';
  --palette-ink-100: '#1a1a1a';
  --palette-ink-70: '#4d4d4d';
  --palette-ink-50: '#808080';
  --palette-ink-30: '#b2b2b2';
  --palette-ink-10: '#e5e5e5';
  --palette-ink-5: '#f4f4f4';
  --palette-white: '#ffffff';

  --semantic-page-foreground: var(--palette-ink-100);
  --semantic-page-background: var(--palette-white);
  --semantic-danger-foreground: #a82e39;
  --semantic-danger-background: var(palette-burgundy-10);
  --semantic-warning-foreground: #9f7509;
  --semantic-warning-background: var(--palette-ochre-10);
  --semantic-success-foreground: #316f2a;
  --semantic-success-background: var(--palette-forest-10);

  --semantic-interactive: var(--palette-azure-70);
  --semantic-interactive-foreground: var(--palette-azure-100);
  --semantic-interactive-background: var(--palette-azure-10);
}
