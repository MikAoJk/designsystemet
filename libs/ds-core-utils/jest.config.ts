/* eslint-disable */
export default {
  displayName: 'ds-core-utils',
  preset: '../../jest.preset.js',
  transform: {
    '^.+\\.[tj]sx?$': ['babel-jest', { presets: ['@nrwl/react/babel'] }],
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../coverage/libs/ds-core-utils',
};
