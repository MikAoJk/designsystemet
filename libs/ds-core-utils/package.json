{
  "name": "@skatteetaten/ds-core-utils",
  "groupId": "no.skatteetaten.aurora",
  "version": "0.0.1",
  "publishConfig": {
    "registry": "https://nexus.sits.no/repository/npm-internal/"
  },
  "peerDependencies": {
    "react": ">=17"
  }
}
