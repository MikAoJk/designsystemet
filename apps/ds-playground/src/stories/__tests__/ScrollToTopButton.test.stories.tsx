import { AxePuppeteer } from '@axe-core/puppeteer';
import {
  getVisibilityThresholdDefault,
  ScrollToTopButton,
  ScrollToTopButtonProps,
} from '@skatteetaten/ds-buttons';
import { ExternalLayout } from '@skatteetaten/ds-core-utils';
import { ComponentStory, Meta } from '@storybook/react';
import { toHaveNoViolations } from 'jest-axe';
// @skatteeteaten/ds-core-designtokens er angitt som symlink i package.json
// derfor vil typecheck feile hvis pakken ikke er bygget, derfor bryter vi nx module boundaries her
// eslint-disable-next-line @nrwl/nx/enforce-module-boundaries
import palette from 'libs/ds-core-designtokens/lib/designtokens/palette.json';
import { Page } from 'puppeteer';

import { category } from '../../../.storybook/helpers';
import { webComponent } from '../../../.storybook/webcomponent-decorator';
import {
  screenShotOptions,
  wrapper,
} from './testUtils/puppeteer.testing.utils';
const defaultButtonText = 'Til toppen';

const verifyMatchSnapShot = async (page: Page): Promise<void> => {
  const innerHtml = await page.$eval(wrapper, (el) => el.innerHTML);
  expect(innerHtml).toMatchSnapshot();
};

const verifyMatchImageSnapShot = async (page: Page): Promise<void> => {
  const image = await page.screenshot(screenShotOptions);
  expect(image).toMatchImageSnapshot();
};

const verifyAxeRules = async (page: Page): Promise<void> => {
  const axeResults = await new AxePuppeteer(page).include(wrapper).analyze();
  expect.extend(toHaveNoViolations);
  expect(axeResults).toHaveNoViolations();
};

const verifySnapshotsAndAxeRules = async (page: Page): Promise<void> => {
  await verifyMatchSnapShot(page);
  await verifyMatchImageSnapShot(page);
  await verifyAxeRules(page);
};

export default {
  component: ScrollToTopButton,
  title: 'Tester/ScrollToTopButton',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },

    // Props
    classNames: {
      table: { disable: true },
    },
    visibilityThreshold: {
      table: {
        category: category.props,
        disable: true,
        //type: { summary: 'number' },
        defaultValue: { summary: getVisibilityThresholdDefault() },
      },
    },
    shadowRootNode: { table: { disable: true } },
    children: { table: { disable: true, category: category.props } },
  },
  parameters: {
    backgrounds: {
      default: 'graphite-70',
      values: [
        {
          name: 'graphite-70',
          value: palette[':root,\n:host']['--palette-graphite-70'],
        },
      ],
    },
  },
} as Meta<ScrollToTopButtonProps>;

const Template: ComponentStory<typeof ScrollToTopButton> = (args) => (
  <div className={'noTransition height100vh'} data-test-block>
    <main className={'scrollToTopContainer'} tabIndex={-1}>
      <ExternalLayout />
      <ScrollToTopButton {...args} />
    </main>
  </div>
);

const defaultArgs = {
  visibilityThreshold: 0,
};

// Når ScrollToTopButton har en ref, så får dom button elementet ref forwarded
export const WithRef = Template.bind({});
WithRef.storyName = 'With Ref (FA1)';

WithRef.args = {
  ...defaultArgs,
  ref: (instance: HTMLButtonElement | null): void => {
    if (instance) {
      instance.id = 'dummyIdForwardedFromRef';
    }
  },
};

WithRef.argTypes = {
  ...WithRef.argTypes,
  ref: { table: { disable: false } },
};
WithRef.parameters = {
  async puppeteerTest(page: Page): Promise<void> {
    await verifyMatchSnapShot(page);
    await verifyAxeRules(page);

    const refId = await page.$eval(
      `${wrapper} > main > div > button`,
      (el) => el.id
    );
    expect(refId).toBe('dummyIdForwardedFromRef');
  },
};

// Når MegaButton har en id, så har button-element id
export const WithId = Template.bind({});
WithId.storyName = 'With Id (FA2)';
WithId.args = {
  ...defaultArgs,
  id: 'htmlid',
};
WithId.argTypes = {
  ...WithId.argTypes,
  id: { table: { disable: false } },
};
WithId.parameters = {
  async puppeteerTest(page: Page): Promise<void> {
    await verifyMatchSnapShot(page);
    await verifyAxeRules(page);

    const elementid = await page.$eval(
      `${wrapper} > main > div > button`,
      (el) => el.getAttribute('id')
    );
    expect(elementid).toBe('htmlid');
  },
};

// Når ScrollToTopButton har en custom CSS, så vises custom stil
export const WithCustomCss = Template.bind({});
WithCustomCss.storyName = 'With Custom CSS (FA3)';
WithCustomCss.args = {
  ...defaultArgs,
  className: 'dummyClassname',
};
WithCustomCss.argTypes = {
  ...WithCustomCss.argTypes,
  className: {
    table: { disable: false },
  },
};
WithCustomCss.parameters = {
  async puppeteerTest(page: Page): Promise<void> {
    await verifySnapshotsAndAxeRules(page);

    const classNameAttribute = await page.$eval(
      `${wrapper} > main > div:nth-child(2)`,
      (el) => el.getAttribute('class')
    );
    expect(classNameAttribute).toContain('dummyClassname');
  },
};

// Når ScrollToTopButton har en custom classNames, så vises custom stil på riktig sted
export const WithCustomClassNames = Template.bind({});
WithCustomClassNames.storyName = 'With Custom ClassNames (FA3)';
WithCustomClassNames.args = {
  ...defaultArgs,
  classNames: {
    container: 'dummyClassname',
    button: 'dummyClassname',
    iconContainer: 'dummyClassname',
    icon: 'dummyClassname',
    label: 'dummyClassname',
  },
};
WithCustomClassNames.argTypes = {
  ...WithCustomClassNames.argTypes,
  classNames: {
    table: { disable: false },
  },
};
WithCustomClassNames.parameters = {
  async puppeteerTest(page: Page): Promise<void> {
    await verifySnapshotsAndAxeRules(page);

    const classNameAttribute = await page.$eval(
      `${wrapper}> main > div > button`,
      (el) => el.getAttribute('class')
    );
    expect(classNameAttribute).toContain('dummyClassname');
  },
};

// Når Button har en lang, så har button-element lang
export const WithLang = Template.bind({});
WithLang.storyName = 'With Lang (FA4)';
WithLang.args = {
  ...defaultArgs,
  lang: 'nb',
};
WithLang.argTypes = {
  ...WithLang.argTypes,
  lang: { table: { disable: false } },
};
WithLang.parameters = {
  async puppeteerTest(page: Page): Promise<void> {
    await verifyMatchSnapShot(page);
    await verifyAxeRules(page);

    const langAttribute = await page.$eval(
      `${wrapper} > main > div > button`,
      (el) => el.getAttribute('lang')
    );
    expect(langAttribute).toBe('nb');
  },
};

// Når ScrollToTopButton har dataTestid, så har button-elementet data-testid satt
export const WithDataTestid = Template.bind({});
WithDataTestid.storyName = 'With DataTestid (FA5)';
WithDataTestid.args = {
  ...defaultArgs,
  'data-testid': '123',
};
WithDataTestid.argTypes = {
  ...WithDataTestid.argTypes,
  'data-testid': { table: { disable: false } },
};
WithDataTestid.parameters = {
  async puppeteerTest(page: Page): Promise<void> {
    await verifyMatchSnapShot(page);
    await verifyAxeRules(page);

    const dataTestid = await page.$eval(
      `${wrapper} > main > div > button`,
      (el) => el.getAttribute('data-testid')
    );
    expect(dataTestid).toBe('123');
  },
};

// Når ScrollToTopButton instansieres, så får den riktige default-verdier og rendrer riktig
export const Defaults = Template.bind({});
Defaults.storyName = 'Defaults (A1, A3, B4)';
Defaults.args = {
  ...defaultArgs,
};
Defaults.argTypes = {
  ...Defaults.argTypes,
  visibilityThreshold: { table: { disable: false } },
};
Defaults.parameters = {
  async puppeteerTest(page: Page): Promise<void> {
    await verifyAxeRules(page);

    const buttonElement = await page.$(`${wrapper} > main > div > button`);
    expect(buttonElement).toMatchSnapshot();

    await verifyMatchSnapShot(page);

    const textContent = await buttonElement?.getProperty('textContent');
    const text = await textContent?.jsonValue();
    expect(text).toBe(defaultButtonText);

    await buttonElement?.focus();
    const imageFocused = await page.screenshot(screenShotOptions);
    expect(imageFocused).toMatchImageSnapshot();

    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    await page.$eval(`${wrapper} > main > div > button`, (el: any) =>
      el.blur()
    );

    await buttonElement?.hover();
    const imageHovered = await page.screenshot(screenShotOptions);
    expect(imageHovered).toMatchImageSnapshot();

    await buttonElement?.click();
    const imageClicked = await page.screenshot(screenShotOptions);
    expect(imageClicked).toMatchImageSnapshot();
  },
};

// Når ScrollToTopButton har children, så settes teksten i button-elementet
export const WithChildren = Template.bind({});
WithChildren.storyName = 'With Children(A2)';
WithChildren.args = {
  ...defaultArgs,
  children: 'dummy string',
};
WithChildren.argTypes = {
  ...WithChildren.argTypes,
  children: { table: { disable: false } },
};
WithChildren.parameters = {
  async puppeteerTest(page: Page): Promise<void> {
    await verifyMatchImageSnapShot(page);
    await verifyAxeRules(page);

    const buttonElement = await page.$(`${wrapper} > main > div > button`);
    const textContent = await buttonElement?.getProperty('textContent');
    const text = await textContent?.jsonValue();
    expect(text).toBe('dummy string');
  },
};

// Når ScrollToTopButton lastes i en small skjerm så får den riktig plassering
export const WithMobileScreen = Template.bind({});
WithMobileScreen.storyName = 'With Small Screen (A5)';
WithMobileScreen.args = {
  ...defaultArgs,
};
WithMobileScreen.parameters = {
  viewport: {
    defaultViewport: '--breakpoint-xs',
  },
  async puppeteerTest(page: Page): Promise<void> {
    await verifyMatchImageSnapShot(page);
    await verifyAxeRules(page);
  },
};

// Når ScrollToTopButton lastes i en bred skjerm så får den riktig plassering
export const WithWideScreen = Template.bind({});
WithWideScreen.storyName = 'With Wide Screen (A6)';
WithWideScreen.args = {
  ...defaultArgs,
};
WithWideScreen.parameters = {
  viewport: {
    defaultViewport: '--breakpoint-xl',
  },
  async puppeteerTest(page: Page): Promise<void> {
    await verifyMatchImageSnapShot(page);
    await verifyAxeRules(page);
  },
};

// Når ScrollToTopButton har visibilityThreshold så vises/skjules knappen ved riktig innslagspunkt for scroll
export const WithVisibilityThreshold = Template.bind({});
WithVisibilityThreshold.storyName =
  'With VisibilityThreshold and Scrolling (A7, A8)';
WithVisibilityThreshold.args = {
  ...defaultArgs,
  visibilityThreshold: 3,
};
WithVisibilityThreshold.argTypes = {
  ...WithVisibilityThreshold.argTypes,
  visibilityThreshold: { table: { disable: false } },
};
WithVisibilityThreshold.parameters = {
  viewport: {
    defaultViewport: '--breakpoint-xl',
  },
  async puppeteerTest(page: Page): Promise<void> {
    await verifyAxeRules(page);

    const className = await page.$eval(
      `${wrapper} > main > div > button`,
      (el) => el.className
    );
    expect(className.match('visible')).toBeFalsy();

    const imageAtTop = await page.screenshot(screenShotOptions);
    expect(imageAtTop).toMatchImageSnapshot();

    await page.evaluate(() => {
      window.scrollBy(0, 3);
    });

    const classNameScrolledDown = await page.$eval(
      `${wrapper} > main > div > button`,
      (el) => el.className
    );
    expect(classNameScrolledDown.match('visible')).toBeTruthy();

    const imageScrolledDown = await page.screenshot(screenShotOptions);
    expect(imageScrolledDown).toMatchImageSnapshot();

    await page.evaluate(() => {
      window.scrollBy(0, -3);
    });

    const classNameScrolledUp = await page.$eval(
      `${wrapper}> main > div > button`,
      (el) => el.className
    );
    expect(classNameScrolledUp.match('visible')).toBeFalsy();

    const imageScrolledUp = await page.screenshot(screenShotOptions);
    expect(imageScrolledUp).toMatchImageSnapshot();
  },
};

const TemplateWithShadowDom: ComponentStory<typeof ScrollToTopButton> = (
  args
) => {
  const element = document.querySelector('scrolltotop-customelement');
  const shadowRoot = element?.shadowRoot;
  return (
    <div className={'noTransition height100vh'} data-test-block>
      <main className={'scrollToTopContainer'} tabIndex={-1}>
        <ExternalLayout />
        <ScrollToTopButton {...args} shadowRootNode={shadowRoot ?? undefined} />
      </main>
    </div>
  );
};
// Når ScrollToTopButton lastes i en shadowDom så tegnes den riktig og klarer å sette focus til main elementet som befinner i et custom element
export const WithShadowDom = TemplateWithShadowDom.bind({});
WithShadowDom.storyName = 'With ShadowDom (B4)';
WithShadowDom.args = {
  ...defaultArgs,
};
WithShadowDom.decorators = [webComponent];
WithShadowDom.parameters = {
  customElementName: 'scrolltotop-customelement',
  async puppeteerTest(page: Page): Promise<void> {
    const customElement = await page.$(`scrolltotop-customelement`);
    const buttonElement = await customElement?.$(
      `pierce/${wrapper} > main > div > button`
    );

    expect(buttonElement).not.toBeNull();
    await buttonElement?.click();
    await page.waitForSelector(`pierce/main:focus`);
  },
};

// Innbakte tekster bruker riktig key
export const WithTranslation = Template.bind({});
WithTranslation.storyName = 'With Translation (A4)';

WithTranslation.args = {
  ...defaultArgs,
};
WithTranslation.argTypes = {
  ...WithTranslation.argTypes,
};
WithTranslation.parameters = {
  locale: 'cimode',
  async puppeteerTest(page: Page): Promise<void> {
    await verifySnapshotsAndAxeRules(page);
    const label = await page.$eval(
      `${wrapper} > main > div > button`,
      (el) => el.textContent
    );
    expect(label).toBe('scrolltotopbutton.Title');
  },
};
