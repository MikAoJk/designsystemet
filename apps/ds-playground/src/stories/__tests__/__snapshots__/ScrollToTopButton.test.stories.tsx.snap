// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/ScrollToTopButton Defaults play-test 1`] = `
<div class="noTransition height100vh"
     data-test-block="true"
>
  <main class="scrollToTopContainer"
        tabindex="-1"
  >
    <div class="ExternalLayout_layout__6aamo ">
      <style>
        div.css-1gh4yjl { max-width: none}
      </style>
      <p>
        <strong>
          Responsive container
        </strong>
      </p>
      <article class="ExternalLayout_article__C2sYl">
        <p>
          <strong>
            Responsive article
          </strong>
        </p>
        <p>
          (uses --semantic-responsive-article)
        </p>
      </article>
      <p>
        (uses --semantic-responsive-container)
      </p>
    </div>
    <div class="ScrollToTopButton_container__5g5Cd  ">
      <button class="ScrollToTopButton_scrollToTopButton__wDr9Z ScrollToTopButton_scrollToTopButton_visible__ulaTu "
              type="button"
      >
        <div class="ScrollToTopButton_scrollToTopButtonIconContainer__sisfC ">
          <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_large__c527o ScrollToTopButton_scrollToTopButtonIconContainerIcon__wAtue "
               viewbox="0 0 24 24"
               focusable="false"
               aria-hidden="true"
               role="img"
          >
            <path d="M11.1 17.6h1.8v1.1h-1.8v-1.1zm0-3.3h1.8v2.2h-1.8v-2.2zm0 6.2h1.8v.5h-1.8v-.5zm-4.9-7.3 5-5v5.1H13V8.2l5 4.9 1.3-1.3-7.3-7-7.1 7.1 1.3 1.3zM21 3H3v1.8h18V3z">
            </path>
          </svg>
        </div>
        <div class="ScrollToTopButton_scrollToTopButtonLabel__vLTWQ ">
          Til toppen
        </div>
      </button>
    </div>
  </main>
</div>
`;

exports[`Tester/ScrollToTopButton WithAttributes play-test 1`] = `
<div class="noTransition height100vh"
     data-test-block="true"
>
  <main class="scrollToTopContainer"
        tabindex="-1"
  >
    <div class="ExternalLayout_layout__6aamo ">
      <style>
        div.css-1gh4yjl { max-width: none}
      </style>
      <p>
        <strong>
          Responsive container
        </strong>
      </p>
      <article class="ExternalLayout_article__C2sYl">
        <p>
          <strong>
            Responsive article
          </strong>
        </p>
        <p>
          (uses --semantic-responsive-article)
        </p>
      </article>
      <p>
        (uses --semantic-responsive-container)
      </p>
    </div>
    <div class="ScrollToTopButton_container__5g5Cd dummyClassname ">
      <button id="htmlId"
              class="ScrollToTopButton_scrollToTopButton__wDr9Z ScrollToTopButton_scrollToTopButton_visible__ulaTu "
              lang="nb"
              type="button"
              data-testid="123ID"
      >
        <div class="ScrollToTopButton_scrollToTopButtonIconContainer__sisfC ">
          <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_large__c527o ScrollToTopButton_scrollToTopButtonIconContainerIcon__wAtue "
               viewbox="0 0 24 24"
               focusable="false"
               aria-hidden="true"
               role="img"
          >
            <path d="M11.1 17.6h1.8v1.1h-1.8v-1.1zm0-3.3h1.8v2.2h-1.8v-2.2zm0 6.2h1.8v.5h-1.8v-.5zm-4.9-7.3 5-5v5.1H13V8.2l5 4.9 1.3-1.3-7.3-7-7.1 7.1 1.3 1.3zM21 3H3v1.8h18V3z">
            </path>
          </svg>
        </div>
        <div class="ScrollToTopButton_scrollToTopButtonLabel__vLTWQ ">
          Til toppen
        </div>
      </button>
    </div>
  </main>
</div>
`;

exports[`Tester/ScrollToTopButton WithChildren play-test 1`] = `
<div class="noTransition height100vh"
     data-test-block="true"
>
  <main class="scrollToTopContainer"
        tabindex="-1"
  >
    <div class="ExternalLayout_layout__6aamo ">
      <style>
        div.css-1gh4yjl { max-width: none}
      </style>
      <p>
        <strong>
          Responsive container
        </strong>
      </p>
      <article class="ExternalLayout_article__C2sYl">
        <p>
          <strong>
            Responsive article
          </strong>
        </p>
        <p>
          (uses --semantic-responsive-article)
        </p>
      </article>
      <p>
        (uses --semantic-responsive-container)
      </p>
    </div>
    <div class="ScrollToTopButton_container__5g5Cd  ">
      <button class="ScrollToTopButton_scrollToTopButton__wDr9Z ScrollToTopButton_scrollToTopButton_visible__ulaTu "
              type="button"
      >
        <div class="ScrollToTopButton_scrollToTopButtonIconContainer__sisfC ">
          <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_large__c527o ScrollToTopButton_scrollToTopButtonIconContainerIcon__wAtue "
               viewbox="0 0 24 24"
               focusable="false"
               aria-hidden="true"
               role="img"
          >
            <path d="M11.1 17.6h1.8v1.1h-1.8v-1.1zm0-3.3h1.8v2.2h-1.8v-2.2zm0 6.2h1.8v.5h-1.8v-.5zm-4.9-7.3 5-5v5.1H13V8.2l5 4.9 1.3-1.3-7.3-7-7.1 7.1 1.3 1.3zM21 3H3v1.8h18V3z">
            </path>
          </svg>
        </div>
        <div class="ScrollToTopButton_scrollToTopButtonLabel__vLTWQ ">
          dummy string
        </div>
      </button>
    </div>
  </main>
</div>
`;

exports[`Tester/ScrollToTopButton WithCustomClassNames play-test 1`] = `
<div class="noTransition height100vh"
     data-test-block="true"
>
  <main class="scrollToTopContainer"
        tabindex="-1"
  >
    <div class="ExternalLayout_layout__6aamo ">
      <style>
        div.css-1gh4yjl { max-width: none}
      </style>
      <p>
        <strong>
          Responsive container
        </strong>
      </p>
      <article class="ExternalLayout_article__C2sYl">
        <p>
          <strong>
            Responsive article
          </strong>
        </p>
        <p>
          (uses --semantic-responsive-article)
        </p>
      </article>
      <p>
        (uses --semantic-responsive-container)
      </p>
    </div>
    <div class="ScrollToTopButton_container__5g5Cd  dummyClassname">
      <button class="ScrollToTopButton_scrollToTopButton__wDr9Z ScrollToTopButton_scrollToTopButton_visible__ulaTu dummyClassname"
              type="button"
      >
        <div class="ScrollToTopButton_scrollToTopButtonIconContainer__sisfC dummyClassname">
          <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_large__c527o ScrollToTopButton_scrollToTopButtonIconContainerIcon__wAtue dummyClassname"
               viewbox="0 0 24 24"
               focusable="false"
               aria-hidden="true"
               role="img"
          >
            <path d="M11.1 17.6h1.8v1.1h-1.8v-1.1zm0-3.3h1.8v2.2h-1.8v-2.2zm0 6.2h1.8v.5h-1.8v-.5zm-4.9-7.3 5-5v5.1H13V8.2l5 4.9 1.3-1.3-7.3-7-7.1 7.1 1.3 1.3zM21 3H3v1.8h18V3z">
            </path>
          </svg>
        </div>
        <div class="ScrollToTopButton_scrollToTopButtonLabel__vLTWQ dummyClassname">
          Til toppen
        </div>
      </button>
    </div>
  </main>
</div>
`;

exports[`Tester/ScrollToTopButton WithMobileScreen smoke-test 1`] = `
<div class="noTransition height100vh"
     data-test-block="true"
>
  <main class="scrollToTopContainer"
        tabindex="-1"
  >
    <div class="ExternalLayout_layout__6aamo ">
      <style>
        div.css-1gh4yjl { max-width: none}
      </style>
      <p>
        <strong>
          Responsive container
        </strong>
      </p>
      <article class="ExternalLayout_article__C2sYl">
        <p>
          <strong>
            Responsive article
          </strong>
        </p>
        <p>
          (uses --semantic-responsive-article)
        </p>
      </article>
      <p>
        (uses --semantic-responsive-container)
      </p>
    </div>
    <div class="ScrollToTopButton_container__5g5Cd  ">
      <button class="ScrollToTopButton_scrollToTopButton__wDr9Z ScrollToTopButton_scrollToTopButton_visible__ulaTu "
              type="button"
      >
        <div class="ScrollToTopButton_scrollToTopButtonIconContainer__sisfC ">
          <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_large__c527o ScrollToTopButton_scrollToTopButtonIconContainerIcon__wAtue "
               viewbox="0 0 24 24"
               focusable="false"
               aria-hidden="true"
               role="img"
          >
            <path d="M11.1 17.6h1.8v1.1h-1.8v-1.1zm0-3.3h1.8v2.2h-1.8v-2.2zm0 6.2h1.8v.5h-1.8v-.5zm-4.9-7.3 5-5v5.1H13V8.2l5 4.9 1.3-1.3-7.3-7-7.1 7.1 1.3 1.3zM21 3H3v1.8h18V3z">
            </path>
          </svg>
        </div>
        <div class="ScrollToTopButton_scrollToTopButtonLabel__vLTWQ ">
          Til toppen
        </div>
      </button>
    </div>
  </main>
</div>
`;

exports[`Tester/ScrollToTopButton WithRef play-test 1`] = `
<div class="noTransition height100vh"
     data-test-block="true"
>
  <main class="scrollToTopContainer"
        tabindex="-1"
  >
    <div class="ExternalLayout_layout__6aamo ">
      <style>
        div.css-1gh4yjl { max-width: none}
      </style>
      <p>
        <strong>
          Responsive container
        </strong>
      </p>
      <article class="ExternalLayout_article__C2sYl">
        <p>
          <strong>
            Responsive article
          </strong>
        </p>
        <p>
          (uses --semantic-responsive-article)
        </p>
      </article>
      <p>
        (uses --semantic-responsive-container)
      </p>
    </div>
    <div class="ScrollToTopButton_container__5g5Cd  ">
      <button class="ScrollToTopButton_scrollToTopButton__wDr9Z ScrollToTopButton_scrollToTopButton_visible__ulaTu "
              type="button"
              id="dummyIdForwardedFromRef"
      >
        <div class="ScrollToTopButton_scrollToTopButtonIconContainer__sisfC ">
          <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_large__c527o ScrollToTopButton_scrollToTopButtonIconContainerIcon__wAtue "
               viewbox="0 0 24 24"
               focusable="false"
               aria-hidden="true"
               role="img"
          >
            <path d="M11.1 17.6h1.8v1.1h-1.8v-1.1zm0-3.3h1.8v2.2h-1.8v-2.2zm0 6.2h1.8v.5h-1.8v-.5zm-4.9-7.3 5-5v5.1H13V8.2l5 4.9 1.3-1.3-7.3-7-7.1 7.1 1.3 1.3zM21 3H3v1.8h18V3z">
            </path>
          </svg>
        </div>
        <div class="ScrollToTopButton_scrollToTopButtonLabel__vLTWQ ">
          Til toppen
        </div>
      </button>
    </div>
  </main>
</div>
`;

exports[`Tester/ScrollToTopButton WithShadowDom play-test 1`] = `
<scrolltotop-customelement>
</scrolltotop-customelement>
`;

exports[`Tester/ScrollToTopButton WithVisibilityThreshold smoke-test 1`] = `
<div class="noTransition height100vh"
     data-test-block="true"
>
  <main class="scrollToTopContainer"
        tabindex="-1"
  >
    <div class="ExternalLayout_layout__6aamo ">
      <style>
        div.css-1gh4yjl { max-width: none}
      </style>
      <p>
        <strong>
          Responsive container
        </strong>
      </p>
      <article class="ExternalLayout_article__C2sYl">
        <p>
          <strong>
            Responsive article
          </strong>
        </p>
        <p>
          (uses --semantic-responsive-article)
        </p>
      </article>
      <p>
        (uses --semantic-responsive-container)
      </p>
    </div>
    <div class="ScrollToTopButton_container__5g5Cd  ">
      <button class="ScrollToTopButton_scrollToTopButton__wDr9Z  "
              type="button"
      >
        <div class="ScrollToTopButton_scrollToTopButtonIconContainer__sisfC ">
          <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_large__c527o ScrollToTopButton_scrollToTopButtonIconContainerIcon__wAtue "
               viewbox="0 0 24 24"
               focusable="false"
               aria-hidden="true"
               role="img"
          >
            <path d="M11.1 17.6h1.8v1.1h-1.8v-1.1zm0-3.3h1.8v2.2h-1.8v-2.2zm0 6.2h1.8v.5h-1.8v-.5zm-4.9-7.3 5-5v5.1H13V8.2l5 4.9 1.3-1.3-7.3-7-7.1 7.1 1.3 1.3zM21 3H3v1.8h18V3z">
            </path>
          </svg>
        </div>
        <div class="ScrollToTopButton_scrollToTopButtonLabel__vLTWQ ">
          Til toppen
        </div>
      </button>
    </div>
  </main>
</div>
`;

exports[`Tester/ScrollToTopButton WithWideScreen smoke-test 1`] = `
<div class="noTransition height100vh"
     data-test-block="true"
>
  <main class="scrollToTopContainer"
        tabindex="-1"
  >
    <div class="ExternalLayout_layout__6aamo ">
      <style>
        div.css-1gh4yjl { max-width: none}
      </style>
      <p>
        <strong>
          Responsive container
        </strong>
      </p>
      <article class="ExternalLayout_article__C2sYl">
        <p>
          <strong>
            Responsive article
          </strong>
        </p>
        <p>
          (uses --semantic-responsive-article)
        </p>
      </article>
      <p>
        (uses --semantic-responsive-container)
      </p>
    </div>
    <div class="ScrollToTopButton_container__5g5Cd  ">
      <button class="ScrollToTopButton_scrollToTopButton__wDr9Z ScrollToTopButton_scrollToTopButton_visible__ulaTu "
              type="button"
      >
        <div class="ScrollToTopButton_scrollToTopButtonIconContainer__sisfC ">
          <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_large__c527o ScrollToTopButton_scrollToTopButtonIconContainerIcon__wAtue "
               viewbox="0 0 24 24"
               focusable="false"
               aria-hidden="true"
               role="img"
          >
            <path d="M11.1 17.6h1.8v1.1h-1.8v-1.1zm0-3.3h1.8v2.2h-1.8v-2.2zm0 6.2h1.8v.5h-1.8v-.5zm-4.9-7.3 5-5v5.1H13V8.2l5 4.9 1.3-1.3-7.3-7-7.1 7.1 1.3 1.3zM21 3H3v1.8h18V3z">
            </path>
          </svg>
        </div>
        <div class="ScrollToTopButton_scrollToTopButtonLabel__vLTWQ ">
          Til toppen
        </div>
      </button>
    </div>
  </main>
</div>
`;
